src/click/types.py

```diff
@@ -329,6 +330,23 @@ def shell_complete(
         return [CompletionItem(c) for c in matched]
 
 
+class EnumChoice(Choice):
+    def __init__(self, enum_type: t.Type[enum.Enum], case_sensitive: bool = True):
+        super().__init__(
+            choices=[element.name for element in enum_type],
```

Comment:
Consider including aliases in this; either as an explicit option (`include_aliases: bool = False`) or by default. Loop over the `enum_type.__members__` mapping to get all names including aliases:

```suggestion
            choices=[name for name in enum_type.__members__],
```

Reply:
Won't that break things like the `case_sensitive` flag?
Like, if you have something like:
```py
import enum

class MyEnum(enum.Enum):
    Foo = 0
    Bar = 1
    Baz = 1  # Alias for `Bar`

# Inside `EnumChoice`:
choices = ["Foo", "Bar"]
aliases = {"Baz": "Bar"}

# And in the `EnumChoice.convert` method:
value = "baz"  # User input
value = aliases.get(value, value)  # "baz" not in `aliases`
# `value` is still "Baz"
# When passed to `super().convert(...)`, it won't know what to do with it
```
I guess this would either require reimplementing the `case_sensitive` handling inside `EnumChoice.convert`, or actually having all possible values, including aliases, in `EnumChoice.choices`.

In my opinion, reimplementing the flag would be sketchy, so I'd probably suggest having all aliases in the choices.

One thing that might be possible, but I have no idea if it actually is, would be to then modify how the argument documentation is generated, to either omit the aliases or to document them together with the primary choice.

Reply:
Not having aliases in `EnumChoice.choices` would also break shell completion for the aliases. This may or may not be undesirable.

Reply:
Another thing, on your suggested change you use:
```py
choices=[name for name in enum_type.__members__],
```
But the following would also work:
```py
choices=list(enum_type.__members__)
```
There's also `enum_type.__members__.keys()`, but that's not a sequence, it's just iterable.

Reply:
*Doh*, yes, `list(enum_type.__members__)` is the obvious choice. I also don't use the case-sensitive option so I indeed missed that.

In that case, case-fold the aliases in the mapping:

```python
        # ...
        self.enum_type = enum_type
        self.aliases = {
            alias.casefold() if case_sensitive else alias: enum_type[alias].name
            for alias in set(enum_type.__members__).difference(self.choices)
        }
        
    def convert(
        self, value: t.Any, param: t.Optional["Parameter"], ctx: t.Optional["Context"]
    ) -> t.Any:
        value = self.aliases.get(value.casefold() if self.case_sensitive else value, value)
        value = super().convert(value=value, param=param, ctx=ctx)
        # ...
```

This does start to get somewhat repetitive with the code in `Choice`, and I did ignore the optional `ctxt.token_normalize_func()` function. Perhaps the normalisation (case folding, `ctx.token_normalize_func()` support) could be factored out into a supporting function that can then be reused here. Something that generates a transformation function based on the `case_sensitive` flag and `ctx`.

Reply:
Not having all values in `choices` will also break `shell_complete` I believe.

Reply:
> Not having all values in `choices` will also break `shell_complete` I believe.

No, it won't *break* completion. You won't be able to use the aliases as completion words but that's not necessarily the point of them, but nothing breaks when you try to complete against a partial alias.